### PR TITLE
New baselines for HtmlTagTest and HtmlErrorTest

### PR DESCRIPTION
**Pull request by ajaybhargavb**
This is the last of Html related baseline tests. There shouldn't be any surprises here.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlTagsTest/VoidElementFollowedByCloseTag.stree.txt [hunk 1]
```diff
@@ -1,563 +1,612 @@
...
-            SyntaxKind.Assign;[=];
-            SyntaxKind.Whitespace;[ ];
-            SyntaxKind.Keyword;[true];
-            SyntaxKind.Semicolon;[;];
-            SyntaxKind.NewLine;[LF];
-        MetaCode span - Gen<None> - [}] - SpanEditHandler;Accepts:None - (581:47,0) - Tokens:1
-            SyntaxKind.RightBrace;[}];
-    Markup span - Gen<None> - [LF] - SpanEditHandler;Accepts:Any - (582:47,1) - Tokens:1
-        SyntaxKind.NewLine;[LF];
+RazorDocument - [0..584)::584 - [@{LF<area> </area>var x = true;LF}LF@{LF<base> </base>var x = true;LF}LF@{LF<br> </br>var x = true;LF}LF@{LF<col> </col>var x = true;LF}LF@{LF<command> </command>var x = true;LF}LF@{LF<embed> </embed>var x = true;LF}LF@{LF<hr> </hr>var x = true;LF}LF@{LF<img> </img>var x = true;LF}LF@{LF<input> </input>var x = true;LF}LF@{LF<keygen> </keygen>var x = true;LF}LF@{LF<link> </link>var x = true;LF}LF@{LF<meta> </meta>var x = true;LF}LF@{LF<param> </param>var x = true;LF}LF@{LF<source> </source>var x = true;LF}LF@{LF<track> </track>var x = true;LF}LF@{LF<wbr> </wbr>var x = true;LF}LF]
```
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlTagsTest/VoidElementFollowedByCloseTag.stree.txt]: This looks huge but it's actually totally repetitive. Can just review the first repetition and the very end.

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlErrorTest/ThrowsErrorIfEndTextTagContainsTextAfterName.stree.txt [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+    MarkupTagBlock - [6..21)::15 - [</text foo bar>]
+        MarkupTransition - [6..21)::15 - Gen<None> - SpanEditHandler;Accepts:Any
+            OpenAngle;[<];
+            ForwardSlash;[/];
+            Text;[text];
+            Whitespace;[ ];
+            Text;[foo];
+            Whitespace;[ ];
+            Text;[bar];
+            CloseAngle;[>];
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlErrorTest/ThrowsErrorIfEndTextTagContainsTextAfterName.stree.txt]: This seems a little odd - why are all of these additional tokens part of a transition? Is this just a recovery bug in the parser? Should we log this?

Comment (ajaybhargavb) [reply test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlErrorTest/ThrowsErrorIfEndTextTagContainsTextAfterName.stree.txt]: This already throws an error `(1,9): Error RZ1023: "<text>" and "</text>" tags cannot contain attributes.`. The `*.diag.txt` file was unchanged.

Comment (rynowak) [reply test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlErrorTest/ThrowsErrorIfEndTextTagContainsTextAfterName.stree.txt]: So why is it treated a transition with like 8 things in it?

Comment (ajaybhargavb) [reply test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlErrorTest/ThrowsErrorIfEndTextTagContainsTextAfterName.stree.txt]: Spoke offline. This is a `MarkupTransition` as opposed to a `CSharpTransition`.

**Pull request merged**